CHANGES.rst

```diff
@@ -1,6 +1,269 @@
+  be removed.
+
+* As a reminder, the :data:`.Unicode` and :data:`.Long` trait types are
+  deprecated since Traits 6.0. Please replace uses with :class:`.Str` and
+  :class:`.Int` respectively. To avoid excessive noise in Traits-using
+  projects, Traits does not yet issue deprecation warnings for existing uses of
+  :data:`.Unicode` and :data:`.Long`. Those warnings will be introduced in a
+  future Traits release, prior to the removal of these trait types.
+
+
+Pending deprecations
+~~~~~~~~~~~~~~~~~~~~
+
+In addition to the deprecations listed in the changelog below, some parts of
+the Traits library are not yet formally deprecated, but are likely to be
+deprecated before Traits 7.0. Users should be aware of the following possible
+future changes:
+
+* The :class:`.Either` trait type will eventually be deprecated. Where possible,
+  use :class:`.Union` instead.
+
+* The ``trait_modified`` event trait that's present on all :class:`.HasTraits`
+  subclasses will eventually be removed. Users should not rely on it being
+  present in an object's ``class_traits`` dictionary.
+
+* Trait names starting with ``trait``, ``traits``, ``_trait`` or
+  ``_traits`` may become reserved for use by ETS at some point in the future.
+  Avoid using these names for your own traits.
+
+Detailed PR-by-PR changes
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+More than 150 PRs went into this release. The following people contributed
+code changes for this release:
+
+* Ieva Cernyte
+* Kit Yan Choi
+* Maxime Costalonga
+* Mark Dickinson
+* Matt Hancock
+* Midhun Madhusoodanan
+* Shoeb Mohammed
+* Franklin Ventura
+* Corran Webster
+
+Features
+~~~~~~~~
+
+* Add ``os.PathLike`` support for ``Directory`` traits. (#867)
+* Add ``Union`` trait type. (#779, #1103, #1107, #1116, #1115)
+* Add ``PrefixList`` trait type. (#871)
+* Add ``allow_none`` flag for ``Callable`` trait. (#885)
+* Add support for type annotation. (#904, #1064)
+* Allow mutable values in ``Constant`` trait. (#929)
+* Add ``Map`` and ``PrefixMap`` trait types. (#886, #953, #956, #970)
+* Add ``TraitList`` as the base list object that can perform validation
+  and emit change notifications. (#912, #981, #984, #989, #999, #1003, #1011,
+  #1026, #1009, #1040)
+* Add ``TraitDict`` as the base dict object that can perform validation and
+  emit change notifications. (#913)
+* Add ``TraitSet`` as the base set object that can perform validation and
+  emit change notifications. (#922, #1043)
+* Implement ``observe`` to supersede ``on_trait_change`` for observing trait
+  changes. (#976, #1000, #1007, #1065, #1023, #1066, #1070, #1069, #1067,
+  #1080, #1082, #1079, #1071, #1072, #1075, #1085, #1089, #1078, #1093, #1086,
+  #1077, #1095, #1102, #1108, #1110, #1112, #1117, #1118, #1123, #1125, #1126)
+* Make ``DefaultValue``, ``ValidateTrait`` and ``TraitKind`` available in
+  ``traits.api``. (#857)
+
+Changes
+~~~~~~~
+
+* GUI applications using Traits 6.1 will require TraitsUI >= 7.0
```

Comment:
I could not quite point this change to a specific PR. Now I wonder if was #1033.
Otherwise this item could  be removed from this section given it has now been mentioned in "Notes on upgrading".

Reply:
Yep; I combined the two changelog entries about this change.